airflow/providers/apache/kafka/operators/consume.py

```diff
@@ -161,7 +161,8 @@ def execute(self, context) -> Any:
                 batch_size = self.max_batch_size
 
             msgs = consumer.consume(num_messages=batch_size, timeout=self.poll_timeout)
-            messages_left -= len(msgs)
+            if not self.read_to_end:
+                messages_left -= len(msgs)
```

Comment:
Discussed and proposed in [slack](https://apache-airflow.slack.com/archives/CCZRF2U5A/p1690806712486489?thread_ts=1690712126.666369&cid=CCZRF2U5A)